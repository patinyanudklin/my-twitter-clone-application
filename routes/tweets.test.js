const request = require('supertest')
const express = require('express')

const tweets = require('./tweets')

const app = express()

app.set('views', './views')
app.set('view engine', 'pug')
app.use('/tweets', tweets)

describe('Tweet Routes', function(){
	describe('GET /tweets/', function(){
		describe('when request success', function(){
			it('should return HTTP Status 200 OK', async function(done){
				const response = await request(app).get('/tweets/')
				expect(response.status).toBe(200)
				expect(response.text).toMatchSnapshot()
				done()
			})
		})
	})
	describe('GET /tweets/search', function(){
		describe('when request success', function(){
			it('should return HTTP Status 200 Found', async function(done){
				const response = await request(app).get('/tweets/search')
				expect(response.status).toBe(200)
				done()
			})
		})
	})
	describe('GET /tweets/new', function(){
		describe('when request success', function(){
			it('should return HTTP Status 200 OK', async function(done){
				const response = await request(app).get('/tweets/new')
				expect(response.status).toBe(200)
				expect(response.text).toMatchSnapshot()
				done()
			})
		})
	})
	describe('GET /tweets/:id', function(){
		describe('when request success', function(){
			it('should return HTTP Status 200 OK', async function(done){
				const response = await request(app).get('/tweets/1')
				expect(response.status).toBe(200)
				expect(response.text).toMatchSnapshot()
				done()
			})
		})
	})
	
})