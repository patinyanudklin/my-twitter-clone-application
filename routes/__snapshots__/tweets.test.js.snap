// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet Routes GET /tweets/ when request success should return HTTP Status 200 OK 1`] = `"<html><head><title>Twitter - List</title><meta charset=\\"utf-8\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"https://cdnjs.cloudflare.com/ajax/libs/bulma/0.7.2/css/bulma.css\\"/><script src=\\"https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js\\"></script><script src=\\"/javascripts/twitter.js\\"></script></head><body><main class=\\"container is-fluid\\"><div class=\\"section background\\"><div class=\\"columns\\"><div class=\\"column\\"><div class=\\"image\\"><a href=\\"/\\"><img src=\\"/twitterIcon.jpg\\" alt=\\"nodejs\\" style=\\"width:75px\\"/></a></div></div><div class=\\"column\\"></div><div class=\\"column\\"></div><div class=\\"column\\"></div><div class=\\"column\\"></div></div><div class=\\"columns\\"><div class=\\"column\\"></div><div class=\\"column\\"><h1 class=\\"title\\">Tweet list</h1><div class=\\"field\\" id=\\"search-box\\"><form action=\\"/tweets/search\\" method=\\"POST\\"><div class=\\"field\\"><div class=\\"control\\"><input class=\\"input\\" name=\\"q\\"/></div></div><div class=\\"field\\"><div class=\\"control\\"><button class=\\"button is-link is-fullwidth\\">Search tweet</button></div></div></form><a href=\\"/tweets/new\\"><button class=\\"button is-link is-fullwidth\\">Add new tweet</button></a></div><br/><div class=\\"div\\" id=\\"list\\"><a href=\\"/tweets/1\\"><div class=\\"box\\" style=\\"border-radius: 0px\\"><strong>patinya</strong><small style=\\"padding-left:10px\\">3 Aug 2019</small><div>test Text</div></div></a><a href=\\"/tweets/2\\"><div class=\\"box\\" style=\\"border-radius: 0px\\"><strong>patinya</strong><small style=\\"padding-left:10px\\">5 Aug 2019</small><div>Team 8 is the best!</div></div></a><a href=\\"/tweets/3\\"><div class=\\"box\\" style=\\"border-radius: 0px\\"><strong>patinya</strong><small style=\\"padding-left:10px\\">5 Aug 2019</small><div>Team 8 is the best!</div></div></a><a href=\\"/tweets/4\\"><div class=\\"box\\" style=\\"border-radius: 0px\\"><strong>miraijyu</strong><small style=\\"padding-left:10px\\">5 Aug 2019</small><div>Miu should be the next center!</div></div></a><a href=\\"/tweets/10\\"><div class=\\"box\\" style=\\"border-radius: 0px\\"><strong>tinya</strong><small style=\\"padding-left:10px\\">15 Aug 2019</small><div>8の日は楽しみ</div></div></a></div></div><div class=\\"column\\"></div></div></div></main></body></html>"`;

exports[`Tweet Routes GET /tweets/:id when request success should return HTTP Status 200 OK 1`] = `"<html><head><title>Twitter - Show</title><meta charset=\\"utf-8\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"https://cdnjs.cloudflare.com/ajax/libs/bulma/0.7.2/css/bulma.css\\"/><script src=\\"https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js\\"></script><script src=\\"/javascripts/twitter.js\\"></script></head><body><main class=\\"container is-fluid\\"><div class=\\"section background\\"><div class=\\"columns\\"><div class=\\"column\\"><div class=\\"image\\"><a href=\\"/\\"><img src=\\"/twitterIcon.jpg\\" alt=\\"nodejs\\" style=\\"width:75px\\"/></a></div></div><div class=\\"column\\"></div><div class=\\"column\\"></div><div class=\\"column\\"></div><div class=\\"column\\"></div></div><div class=\\"columns\\"><div class=\\"column\\"></div><div class=\\"column\\"><a href=\\"/tweets/1\\"><div class=\\"box\\" style=\\"border-radius: 0px\\"><strong>patinya</strong><small style=\\"padding-left:10px\\">3 Aug 2019</small><div>test Text</div></div></a></div><div class=\\"column\\"></div></div></div></main></body></html>"`;

exports[`Tweet Routes GET /tweets/new when request success should return HTTP Status 200 OK 1`] = `"<html><head><title>Twitter - New</title><meta charset=\\"utf-8\\"/><link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"https://cdnjs.cloudflare.com/ajax/libs/bulma/0.7.2/css/bulma.css\\"/><script src=\\"https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js\\"></script><script src=\\"/javascripts/twitter.js\\"></script></head><body><main class=\\"container is-fluid\\"><div class=\\"section background\\"><div class=\\"columns\\"><div class=\\"column\\"><div class=\\"image\\"><a href=\\"/\\"><img src=\\"/twitterIcon.jpg\\" alt=\\"nodejs\\" style=\\"width:75px\\"/></a></div></div><div class=\\"column\\"></div><div class=\\"column\\"></div><div class=\\"column\\"></div><div class=\\"column\\"></div></div><div class=\\"columns\\"><div class=\\"column\\"></div><div class=\\"column\\"><h1 class=\\"title\\">New Tweet</h1><form action=\\"/tweets/new\\" method=\\"POST\\"><div class=\\"field\\"><div class=\\"control\\"><input class=\\"input\\" name=\\"tweet\\"/></div></div><div class=\\"field\\"><div class=\\"control\\"><button class=\\"button is-link is-fullwidth\\">Create Tweet</button></div></div></form></div><div class=\\"column\\"></div></div></div></main></body></html>"`;
